File: javascript/eliuds-eggs/eliuds-eggs.js
//
// This is only a SKELETON file for the 'Eliud's Eggs' exercise. It's been provided as a
// convenience to get you started writing code faster.
//

export const eggCount = (displayValue) => {
  let eggs = 0
  let value = displayValue

  while (value !== 0)
  {
    eggs += value & 0b1
    value >>= 1
  }

  return eggs
};

File: csharp/eliuds-eggs/EliudsEggs.cs
public static class EliudsEggs
{
    public static int EggCount(int encodedCount)
    {
        int eggs = 0;
        int encoded = encodedCount;
        while (encoded != 0)
        {
            eggs += encoded & 0b1;
            encoded >>= 1;            
        }
        return eggs;
    }
}
